keanu-project/src/main/java/io/improbable/keanu/vertices/VertexLabel.java

```diff
@@ -1,30 +1,58 @@
 package io.improbable.keanu.vertices;
 
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
 import java.util.Objects;
 
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableList;
+
 public class VertexLabel {
-    private final String label;
+    private static final char NAMESPACE_SEPARATOR = '.';
+    private static final char NAME_PREPENDER = ':';
+    private final String name;
+    private final List<String> namespace;
+
+    public VertexLabel(String name, String... namespace) {
+        this.name = name;
+        this.namespace = ImmutableList.copyOf(namespace);
+    }
+
+    public VertexLabel inNamespace(String topLevelNamespace) {
+        List<String> newNamespace = ImmutableList.<String>builder().addAll(namespace).add(topLevelNamespace).build();
+        return new VertexLabel(this.name, newNamespace.toArray(new String[0]));
```

Comment:
Totally not your doing - but I'm not loving this instance of the toArray API - doesn't seem ideal to have to create an array for no reason!

Reply:
no problem, I'll just add a constructor that takes a List